website/public/js/controllers/tasksCtrl.js

```diff
@@ -24,13 +24,100 @@ habitrpg.controller("TasksCtrl", ['$scope', '$rootScope', '$location', 'User','N
     };
 
     function addTask(addTo, listDef, task) {
+      var checkRegex = function(match, k) {
+        var key = k || 1;
+        if (match && match.length > key && typeof match[key] !== 'undefined') {
+          return match[key];
+        }
+        return;
+      };
+      var regexes = {
+        notes: /(?:\s+--"([^"]+)")/,
+        due: /(?:\s+\^(\d?\d\/\d?\d\/\d?\d?\d\d))/,
+        diff: /(?:\s+\!(easy|medium|hard))/,
+        tag: /(?:\s+\#([^"\s]+|"[^"]+"))/g,
+        value: /(?:\s+(\d+(?:\.\d{1,2})?|0?\.\d{1,2})?GP)/,
+        up: /(?:\s+(\+))/,
+        down: /(?:\s+(\-))/,
```

Comment:
I agree. It's apparent I definitely need to test a lot more cases than the few I've done so far. I agree about keeping the current default.